### ISSUE TITLE
"No messages" when there are messages

### ISSUE THREAD

**Issue by fileoffset** [zulip/zulip-mobile#3162]
On Android, latest client from Store.

Seems to be a new bug/regression. 

Sometimes when opening the app, and then opening a private conversation, the app shows new messages but when you click through to the private chat, no messages appear and an error is shown. 

If you go back, and then re-enter the conversation, the messages reappear.

**Comment by gnprice** [comment]
Thanks @fileoffset for this report! We've seen a couple of similar reports, but hadn't yet made a proper issue to track it. I'll add our other information here so it's in one place.

We have a possible diagnosis for the bug and are working on a fix; see #3151 .

**Comment by gnprice** [comment]
Several people report showing "no messages" in a conversation that definitely has messages, shortly after opening the app.
* Reports of this so far are all on 20.0.103; first one was Nov 14.
* One report was on the "All messages" view.
* In one screenshot, the "1 unread message" banner appears, above "No messages with this person".
* All reports find it's fixed by going back and then re-entering the conversation.

Nov 15, @borisyankov and I [dug into this](https://chat.zulip.org/#narrow/stream/48-mobile/subject/no.20messages/near/661974)...
> with some help from the server logs after he saw it on his account, and I think we have a diagnosis! And a plan to fix.
>
> The sequence when he saw it was:
> * App opens, and goes to a narrow. (He'd tapped a PM notification.)
> * App tries to poll its event queue for updates; server replies it's expired.
> * App invalidates its cache for the messages. This is as designed, because they might have been edited, read, reacted to, etc. Kind of rough to completely discard them, though, because probably most of them didn't change, or change much; we have ideas on how to soften that with some server changes.
> * Then the buggy part: the app doesn't kick off a request to fetch the messages for that narrow.
>
> Oh, meanwhile:
> * App creates a new event queue and gets subsequent events. Everything's working great except we don't know about those messages and aren't going out to learn about them.
> * In fact, within the app's state, all its information is correct: it doesn't know about any messages, but it also knows it doesn't know. (The "caught up" bits get cleared when the cache is invalidated.)
>   * In addition to the fact we're not fetching, this indicates a UI issue: we shouldn't be telling the user "No messages" when in fact we don't know if there are messages. If we don't know if there are messages and yet we aren't actively fetching to see if there are... @borisyankov and I just chatted about this in person and probably we should show a failure message, maybe "Couldn't connect". Once this bug is fixed, that's still a situation we'll land in when a fetch fails.

Boris has started on a fix, #3151 .

**Comment by EmilyRosina** [comment]
This is happening again.

**Comment by gnprice** [comment]
@EmilyRosina Interesting, thanks. Please file a new issue, with details of what's happening when you see this. If you can include a screenshot or screen recording, that would be especially helpful.
